### DISCUSSION TITLE
🍬 Candy Store

### DISCUSSION THREAD

**Discussion by bjlittle** [bjlittle/geovista#1033] [Show and Tell]
We're super excited about `geovista` ... but we're even more excited to find out who's using `geovista` and how you're using it.

Do you have any lush eye candy that you'd 💚 to share with the community? Be it a static plot or an animation, poster or scientific paper?

Consider this a safe space to really go to town and showcase what you're proud and excited about!

Please feel free to add links to your repo, academic journal or paper, app, website or gist etc. This space is your space to share and promote your awesome work to the community.

Go for it! Flex away!

Show us what you've got 🤩

**Comment by dennissergeev** [comment]
To start the ball rolling, I'll re-post the figure I made for my paper on terrestrial exoclimate simulations, which was also the first citation of `geovista` in an academic paper (yay! :tada:).

[Sergeev et al. (2023), Geosci. Model Dev., DOI:10.5194/gmd-16-5601-2023](https://doi.org/10.5194/gmd-16-5601-2023)

![thai_hab1__C48__0500d__t_sfc_vl_l_m_h_vh_cloud_frac](https://github.com/user-attachments/assets/de76402e-60c6-40e4-a873-4a00a12f61ea)

**Comment by dennissergeev** [comment]
Another paper where `geovista` was particularly useful: [Sergeev et al. (2024), ApJ, DOI:10.3847/1538-4357/ad4ecd](https://doi.org/10.3847/1538-4357/ad4ecd). With `geovista`, I was easily able to visualise my simulations that used a stretched mesh.

![Figure 2 from the paper](https://github.com/user-attachments/assets/350d86c5-1911-4085-afa7-882a678e5d43)
**Figure 2.** Clouds and precipitation in the simulations with stretched and quasi-uniform mesh.

Code repository with the plotting scripts for this paper: [dennissergeev/stretched_mesh_code](https://github.com/dennissergeev/stretched_mesh_code).

---

![hab1_mod_c192_p_s10e__cell_geometry](https://github.com/user-attachments/assets/5507450c-8433-45e2-9df3-60caa6bc4359)
**Bonus figure A.** Mesh topology and cell size in the simulations with stretched and quasi-uniform mesh.

![thai_hab1_C192s0p125__0290d__m_h_vh_cloud_frac__stars_and_code_bg_square_v0__zoom180__ready](https://github.com/user-attachments/assets/fe99a7a7-fbc1-4ad0-ba61-5abc2f999de2)
**Bonus figure B.** Stretched-mesh simulations of the cloud cover and temperature on a tidally-locked rocky exoplanet.

**Reply by bjlittle** [reply]
Thanks @dennissergeev! This is amazing!

We're so excited to see `geovista` already used in published scientific papers. Simply incredible! Congrats on bagging the first academic citation of `geovista` 🏆 

I totally ❤️ the stretched-mesh cloud cover and temperature renders above (Bonus figure B). Just gorgeous 😍

**Reply by dennissergeev** [reply]
No, thank _you_ for creating such a great package! :blush:

**Comment by sit23** [comment]
I've loved using Geovista for a variety of projects (albeit outreach and teaching related so far). Recently I've been automatically processing GEFS weather forecast data and making interactive plots of the variables with geovista:
E.g.:
https://gefs-geovista-plot-website.s3.eu-west-2.amazonaws.com/gefs_zarr_store/gefs.20230707/00/atmos/pgrb2ap5/pwat_000_384.html

I've also used geovista to make animations of these fields, e.g.:
https://gefs-geovista-plot-website.s3.eu-west-2.amazonaws.com/gefs_zarr_store/gefs.20230707/00/atmos/pgrb2ap5/pwat_000_384.mp4

I've mostly used these as demonstrations during my teaching sessions at the University of Exeter and in outreach sessions with school groups, and I'm hoping to use it for some upcoming work I'm doing looking at GCM modelling of Jupiter (a bit like @dennissergeev's Jupiters, but colder!). 

Thanks so much for making this brilliant package!

**Reply by bjlittle** [reply]
@sit23 Awesome! Thanks so much for sharing and using `geovista`! 🤩 

Really looking forward to seeing your work on modelling Jupiter 💯

**Comment by kuchaale** [comment]
I want to share a pyvista example of Hunga-Tonga eruption in our ESM SOCOLv4 (see for details [here](https://egusphere.copernicus.org/preprints/2024/egusphere-2024-1909/)). I hope I will be able to translate it using Geovista in the future.

![time_all_SAD+WV](https://github.com/user-attachments/assets/7dea765a-48d1-4a1f-9eef-05b55888c1c2)

**Reply by bjlittle** [reply]
Ohhh really lovely work @kuchaale !

Thanks for sharing 🤩

**Reply by kuchaale** [reply]
I updated my visualization in spherical coordinates (see below) but still using code based on PyVista adapted from @dennissergeev. I will try again with [🦊 GeoVista v0.6.0](https://github.com/users/bjlittle/projects/9/views/1)

![time_all_WV_geo_202201-202304_withanimation](https://github.com/user-attachments/assets/7e5f4900-5cb1-4426-bf66-3d1e2dd9e3c9)

**Reply by dennissergeev** [reply]
Thanks for sharing @kuchaale! Would you mind sharing (or DM'ing) the code too? No worries if not!

**Reply by kuchaale** [reply]
@dennissergeev [here](https://gist.github.com/kuchaale/b8bd2aaa02a0eba257c8301ce466c767) is the code snippet

**Reply by dennissergeev** [reply]
thanks @kuchaale!

**Reply by jbusecke** [reply]
Hi @kuchaale, is that data available somewhere? Id love to test it with this [PR](https://github.com/bjlittle/geovista/pull/2294).

**Comment by DanCopsey** [comment]
I have written a GUI interface to load a netcdf file and view the contents using Geovista. It is called ngview and it is here:

https://github.com/MetOffice/ngview

<img width="1280" height="720" alt="example_window" src="https://github.com/user-attachments/assets/7c37312c-5763-4050-991a-690b5d519591" />

Feel free to pass around...

**Reply by bjlittle** [reply]
@DanCopsey Brilliant! 

This looks like a super useful tool for scientists to easily explore their data. Nice one!

Congrats 🥳🎉

**Comment by changliao1025** [comment]
We have used GeoVista in our American Geophysical Union Fall meeting 2025. An animated version is below:
https://www.youtube.com/watch?v=i0cGuaZR9Yg

**Reply by bjlittle** [reply]
@changliao1025 Wow! That's a really stunning animation :star_struck: 

Congrats! I totally :heart: it!
